bundles/io/org.eclipse.smarthome.io.json/src/main/java/org/eclipse/smarthome/io/json/internal/gson/OsgiGsonBindingService.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.json.internal.gson;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.io.json.JsonBindingService;
+import org.eclipse.smarthome.io.json.gson.GsonBindingService;
+import org.eclipse.smarthome.io.json.gson.GsonTypeAdapterProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JSON data binding implementation that uses Google Gson to serialize and deserialize between
+ * Java objects and JSON.
+ *
+ * @author Flavio Costa - Initial contribution and API
+ *
+ * @param <T> Type to be serialized or deserialized.
+ */
+@Component(immediate = true, service = JsonBindingService.class, property = {
+        OsgiGsonBindingService.FORMAT_OUTPUT_PROPERTY + ":Boolean=false" })
+public class OsgiGsonBindingService<T> extends GsonBindingService<T> {
```

Comment:
I would not call it something with `OSGI` in the name, but you cannot simply remove it because the underlying service has already that name. Do you have another naming idea? :)

Reply:
In the workspace there are 24 types with a name starting with `OSGI`, but since you didn't like it I renamed the class as `InternalGsonBindingService`. This is there just to expose the service as an OSGi component, and as the new name suggests, it's an internal implementation in a package not exported by the bundle.